src/duration/mod.rs

```diff
@@ -242,10 +243,51 @@ impl Div<u32> for Duration {
 }
 
 impl Duration {
+    /// Creates a new [`Duration`] from the specified number of whole milliseconds.
+    pub const fn from_millis(millis: u64) -> Self {
+        Self(std::time::Duration::from_millis(millis))
+    }
+
     /// Creates a new [`Duration`] from the specified number of whole seconds.
     pub const fn from_secs(secs: u64) -> Self {
         Self(std::time::Duration::from_secs(secs))
     }
+
+    /// Creates a new [`Duration`] from the specified number of whole minutes.
+    /// Panics if the Duration is higher than [`std::time::Duration`] supports
+    /// (which is 584.9 years). Please only use this function in constant contexts,
+    /// and not based upon user input!
+    pub const fn from_minutes_unchecked(minutes: u64) -> Self {
+        let millis = match minutes.checked_mul(DurationUnit::Minutes.millis()) {
+            Some(millis) => millis,
+            None => panic!("overflow in Duration::from_minutes"),
```

Comment:
Maybe add some more context to the panic message, like with which value (of minutes in this case) the value overflowed. This should make debugging a little more clear.

Reply:
Got this on my try...
```
error[E0015]: cannot call non-const formatting macro in constant functions
   --> src/duration/mod.rs:265:75
    |
265 |             None => panic!("overflow in Duration::from_minutes by passing {minutes} minutes"),
    |                                                                           ^^^^^^^^^
    |
    = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
    = note: this error originates in the macro `$crate::const_format_args` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Ah... That's unfortunate. Let's leave it as is I guess.